### PR TITLE
Added support for GPS and included serial library.

### PR DESCRIPTION
**Pull request by tonybaltovski**

### CODE REVIEW

**Comment by mikepurvis** [comment]
LGTM.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,27 +2,32 @@ cmake_minimum_required(VERSION 2.8.3)
...
+    include
+    ${Boost_INCLUDE_DIRS}
     ${catkin_INCLUDE_DIRS}
 )
 
+add_library(hg_ros_serial src/serial.cpp)
+
 add_executable(3dm_gx4_test src/3dm_gx4_test.cpp src/3dm_gx4.cpp)
-target_link_libraries(3dm_gx4_test  
+target_link_libraries(3dm_gx4_test hg_ros_serial
```
Comment (mikepurvis) [inline CMakeLists.txt:25]: linebreak here? (2 places)

File: include/hg_ros_3dm_gx4/mip.h [hunk 2]
```diff
@@ -157,6 +157,9 @@ struct IMUData
     NUM_IMU_DATA = 12
   };
 
+  static const float GAUSS_TO_TESLA = 0.0001;
+  static const float G_TO_ACCLERATION = 9.81;
```
Comment (mikepurvis) [inline include/hg_ros_3dm_gx4/mip.h]: sp

File: include/hg_ros_3dm_gx4/mip.h [hunk 3]
```diff
@@ -193,7 +196,27 @@ struct GPSData
...
+  float vel[3];
+  float vel_accuracy;
+  uint16_t year;
+  uint8_t date[5];
+  uint32_t mills;
+  uint32_t posix_time;
+
+
+  GPSData()
+    : fields(0)
```
Comment (mikepurvis) [inline include/hg_ros_3dm_gx4/mip.h:216]: Nothing else to initialize here?

File: src/3dm_gx4_node.cpp [hunk 4]
```diff
@@ -4,67 +4,210 @@
...
+ros::Publisher g_pub_filtered_fix;
+ros::Publisher g_pub_filtered_vel;
+sensor_msgs::Imu g_raw_imu;
+sensor_msgs::MagneticField g_raw_mag;
+sensor_msgs::NavSatFix g_navsat_fix;
+geometry_msgs::TwistStamped g_navsat_vel;
+sensor_msgs::TimeReference g_navsat_tf;
+sensor_msgs::Imu g_filtered_imu;
+sensor_msgs::NavSatFix g_filtered_fix;
+geometry_msgs::TwistWithCovarianceStamped g_filtered_vel;
```
Comment (mikepurvis) [inline src/3dm_gx4_node.cpp]: What's the reason for all these messages being global? Couldn't they be local in the functions which do the publishing?

Comment (tonybaltovski) [inline src/3dm_gx4_node.cpp]: Not really, I will make them local.

File: include/hg_ros_3dm_gx4/mip.h (hunk 3 above)
Comment (tonybaltovski) [inline include/hg_ros_3dm_gx4/mip.h:216]: Not really.

**Pull request merged**